nilearn/connectome/group_sparse_cov.py

```diff
@@ -502,7 +502,7 @@ def _group_sparse_covariance(
     return omega
 
 
-class GroupSparseCovariance(BaseEstimator, CacheMixin):
+class GroupSparseCovariance(CacheMixin, BaseEstimator):
```

Comment:
Hust curious: what is the reason for this switch ?

Reply:
this is a recommendation from scikit-learn related to method resolution order in the case of multiple inheritance

> It is particularly important to notice that mixins should be “on the left” while the BaseEstimator should be “on the right” in the inheritance list for proper MRO.

Source: https://scikit-learn.org/dev/developers/develop.html#rolling-your-own-estimator